pkg/reconciler/kubernetes/tektonconfig/extension.go

```diff
@@ -45,9 +46,40 @@ func (oe kubernetesExtension) PreReconcile(context.Context, v1alpha1.TektonCompo
 }
 func (oe kubernetesExtension) PostReconcile(ctx context.Context, comp v1alpha1.TektonComponent) error {
 	configInstance := comp.(*v1alpha1.TektonConfig)
-	if configInstance.Spec.Profile == common.ProfileAll {
+
+	switch configInstance.Spec.Profile {
+	case common.ProfileAll:
 		return extension.CreateDashboardCR(comp, oe.operatorClientSet.OperatorV1alpha1())
+
+	case common.ProfileBasic:
+		_, err := extension.GetDashboard(oe.operatorClientSet.OperatorV1alpha1().TektonDashboards(), common.DashboardResourceName)
+		if err != nil {
+			return err
+		}
+		return extension.TektonDashboardCRDelete(oe.operatorClientSet.OperatorV1alpha1().TektonDashboards(), common.DashboardResourceName)
+
+	case common.ProfileLite:
+		dashboard, err := extension.GetDashboard(oe.operatorClientSet.OperatorV1alpha1().TektonDashboards(), common.DashboardResourceName)
+		if dashboard != nil {
```

Comment:
```suggestion
			if errors.IsNotFound(err) {
   			    return nil
			}
```
i think a check whether there is no error should be enough to decide to delete